### PR TITLE
WIP: python3Packages.torch: migrate to setupCudaHook

### PR DESCRIPTION
**Pull request by SomeoneSerge**
- python3Packages.torch: asserts -> meta.broken
- python3Packages.torch: symlinkJoin -> CUDAToolkit_ROOT
- python3Packages.pynvml: drop the unused cudatoolkit input

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/torch/default.nix [hunk 1]
```diff
@@ -52,15 +52,9 @@
...
-  inherit (cudaPackages) cudatoolkit cudaFlags cudnn nccl;
 in
 
-assert cudaSupport -> stdenv.isLinux;
-assert cudaSupport -> (cudaPackages.cudaMajorVersion == "11");
-
-# confirm that cudatoolkits are sync'd across dependencies
-assert !(MPISupport && cudaSupport) || mpi.cudatoolkit == cudatoolkit;
-assert !cudaSupport || magma.cudaPackages.cudatoolkit == cudatoolkit;
+with cudaPackages;
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/torch/default.nix:57]: I probably shouldn't use `with`

**Comment by SomeoneSerge** [comment]
Result of `nixpkgs-review pr 245516 --extra-nixpkgs-config '{ cudaCapabilities = [ "8.6" ]; cudaSupport = true; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.pymanopt</li>
    <li>python310Packages.pymanopt.dist</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-machine-learning</li>
    <li>python310Packages.qiskit-machine-learning.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.pymanopt</li>
    <li>python311Packages.pymanopt.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
  </ul>
</details>
<details>
  <summary>174 packages failed to build:</summary>
  <ul>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>piper-train (python310Packages.piper-train)</li>
    <li>piper-train.dist (python310Packages.piper-train.dist)</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.clean-fid</li>
    <li>python310Packages.clean-fid.dist</li>
    <li>python310Packages.cleanlab</li>
    <li>python310Packages.cleanlab.dist</li>
    <li>python310Packages.clip</li>
    <li>python310Packages.clip-anytorch</li>
    <li>python310Packages.clip-anytorch.dist</li>
    <li>python310Packages.clip.dist</li>
    <li>python310Packages.detectron2</li>
    <li>python310Packages.detectron2.dist</li>
    <li>python310Packages.effdet</li>
    <li>python310Packages.effdet.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.ffcv.dist</li>
    <li>python310Packages.grad-cam</li>
    <li>python310Packages.grad-cam.dist</li>
    <li>python310Packages.invisible-watermark</li>
    <li>python310Packages.invisible-watermark.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.manifest-ml</li>
    <li>python310Packages.manifest-ml.dist</li>
    <li>python310Packages.mmcv</li>
    <li>python310Packages.mmcv.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pytorch-lightning</li>
    <li>python310Packages.pytorch-lightning.dist</li>
    <li>python310Packages.pytorch-metric-learning</li>
    <li>python310Packages.pytorch-metric-learning.dist</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.pytorch-pfn-extras.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.tensorboardx</li>
    <li>python310Packages.tensorboardx.dist</li>
    <li>python310Packages.timm</li>
    <li>python310Packages.timm.dist</li>
    <li>python310Packages.tiny-cuda-nn</li>
    <li>python310Packages.torch-tb-profiler</li>
    <li>python310Packages.torch-tb-profiler.dist</li>
    <li>python310Packages.torchinfo</li>
    <li>python310Packages.torchinfo.dist</li>
    <li>python310Packages.torchvision</li>
    <li>python310Packages.torchvision.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python310Packages.xformers</li>
    <li>python310Packages.xformers.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>python311Packages.apptools</li>
    <li>python311Packages.apptools.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.ax</li>
    <li>python311Packages.ax.dist</li>
    <li>python311Packages.botorch</li>
    <li>python311Packages.botorch.dist</li>
    <li>python311Packages.clean-fid</li>
    <li>python311Packages.clean-fid.dist</li>
    <li>python311Packages.cleanlab</li>
    <li>python311Packages.cleanlab.dist</li>
    <li>python311Packages.clip</li>
    <li>python311Packages.clip-anytorch</li>
    <li>python311Packages.clip-anytorch.dist</li>
    <li>python311Packages.clip.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.detectron2</li>
    <li>python311Packages.detectron2.dist</li>
    <li>python311Packages.easyocr</li>
    <li>python311Packages.easyocr.dist</li>
    <li>python311Packages.effdet</li>
    <li>python311Packages.effdet.dist</li>
    <li>python311Packages.envisage</li>
    <li>python311Packages.envisage.dist</li>
    <li>python311Packages.fairseq</li>
    <li>python311Packages.fairseq.dist</li>
    <li>python311Packages.fastai</li>
    <li>python311Packages.fastai.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.grad-cam</li>
    <li>python311Packages.grad-cam.dist</li>
    <li>python311Packages.invisible-watermark</li>
    <li>python311Packages.invisible-watermark.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.mayavi</li>
    <li>python311Packages.mayavi.dist</li>
    <li>python311Packages.mmcv</li>
    <li>python311Packages.mmcv.dist</li>
    <li>python311Packages.mmengine</li>
    <li>python311Packages.mmengine.dist</li>
    <li>python311Packages.nianet</li>
    <li>python311Packages.nianet.dist</li>
    <li>python311Packages.pytorch-lightning</li>
    <li>python311Packages.pytorch-lightning.dist</li>
    <li>python311Packages.pytorch-metric-learning</li>
    <li>python311Packages.pytorch-metric-learning.dist</li>
    <li>python311Packages.pytorch-pfn-extras</li>
    <li>python311Packages.pytorch-pfn-extras.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.skorch</li>
    <li>python311Packages.skorch.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.spacy</li>
    <li>python311Packages.spacy-loggers</li>
    <li>python311Packages.spacy-loggers.dist</li>
    <li>python311Packages.spacy-lookups-data</li>
    <li>python311Packages.spacy-lookups-data.dist</li>
    <li>python311Packages.spacy-transformers</li>
    <li>python311Packages.spacy-transformers.dist</li>
    <li>python311Packages.spacy.dist</li>
    <li>python311Packages.tensorboardx</li>
    <li>python311Packages.tensorboardx.dist</li>
    <li>python311Packages.textacy</li>
    <li>python311Packages.textacy.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>python311Packages.timm</li>
    <li>python311Packages.timm.dist</li>
    <li>python311Packages.tiny-cuda-nn</li>
    <li>python311Packages.torch-tb-profiler</li>
    <li>python311Packages.torch-tb-profiler.dist</li>
    <li>python311Packages.torchinfo</li>
    <li>python311Packages.torchinfo.dist</li>
    <li>python311Packages.torchvision</li>
    <li>python311Packages.torchvision.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
    <li>python311Packages.wandb</li>
    <li>python311Packages.wandb.dist</li>
    <li>python311Packages.xformers</li>
    <li>python311Packages.xformers.dist</li>
  </ul>
</details>
<details>
  <summary>246 packages built:</summary>
  <ul>
    <li>gwe</li>
    <li>libretranslate (python310Packages.libretranslate)</li>
    <li>libretranslate.dist (python310Packages.libretranslate.dist)</li>
    <li>ocamlPackages.torch</li>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>openai-whisper (python310Packages.openai-whisper)</li>
    <li>openai-whisper.dist (python310Packages.openai-whisper.dist)</li>
    <li>pianotrans</li>
    <li>pianotrans.dist</li>
    <li>python310Packages.accelerate</li>
    <li>python310Packages.accelerate.dist</li>
    <li>python310Packages.apptools</li>
    <li>python310Packages.apptools.dist</li>
    <li>python310Packages.argos-translate-files</li>
    <li>python310Packages.argos-translate-files.dist</li>
    <li>python310Packages.argostranslate</li>
    <li>python310Packages.argostranslate.dist</li>
    <li>python310Packages.ax</li>
    <li>python310Packages.ax.dist</li>
    <li>python310Packages.bitsandbytes</li>
    <li>python310Packages.bitsandbytes.dist</li>
    <li>python310Packages.blosc2</li>
    <li>python310Packages.blosc2.dist</li>
    <li>python310Packages.botorch</li>
    <li>python310Packages.botorch.dist</li>
    <li>python310Packages.ctranslate2</li>
    <li>python310Packages.ctranslate2.dist</li>
    <li>python310Packages.deepdish</li>
    <li>python310Packages.deepdish.dist</li>
    <li>python310Packages.deepwave</li>
    <li>python310Packages.deepwave.dist</li>
    <li>python310Packages.envisage</li>
    <li>python310Packages.envisage.dist</li>
    <li>python310Packages.experiment-utilities</li>
    <li>python310Packages.experiment-utilities.dist</li>
    <li>python310Packages.ezyrb</li>
    <li>python310Packages.ezyrb.dist</li>
    <li>python310Packages.fairscale</li>
    <li>python310Packages.fairscale.dist</li>
    <li>python310Packages.fairseq</li>
    <li>python310Packages.fairseq.dist</li>
    <li>python310Packages.faster-whisper</li>
    <li>python310Packages.faster-whisper.dist</li>
    <li>python310Packages.flammkuchen</li>
    <li>python310Packages.flammkuchen.dist</li>
    <li>python310Packages.flyingsquid</li>
    <li>python310Packages.flyingsquid.dist</li>
    <li>python310Packages.fvcore</li>
    <li>python310Packages.fvcore.dist</li>
    <li>python310Packages.gpytorch</li>
    <li>python310Packages.gpytorch.dist</li>
    <li>python310Packages.ignite</li>
    <li>python310Packages.ignite.dist</li>
    <li>python310Packages.iopath</li>
    <li>python310Packages.iopath.dist</li>
    <li>python310Packages.kornia</li>
    <li>python310Packages.kornia.dist</li>
    <li>python310Packages.layoutparser</li>
    <li>python310Packages.layoutparser.dist</li>
    <li>python310Packages.linear_operator</li>
    <li>python310Packages.linear_operator.dist</li>
    <li>python310Packages.lion-pytorch</li>
    <li>python310Packages.lion-pytorch.dist</li>
    <li>python310Packages.mayavi</li>
    <li>python310Packages.mayavi.dist</li>
    <li>python310Packages.mmengine</li>
    <li>python310Packages.mmengine.dist</li>
    <li>python310Packages.monai</li>
    <li>python310Packages.monai.dist</li>
    <li>python310Packages.nianet</li>
    <li>python310Packages.nianet.dist</li>
    <li>python310Packages.pandas-stubs</li>
    <li>python310Packages.pandas-stubs.dist</li>
    <li>python310Packages.pdfplumber</li>
    <li>python310Packages.pdfplumber.dist</li>
    <li>python310Packages.peft</li>
    <li>python310Packages.peft.dist</li>
    <li>python310Packages.pgmpy</li>
    <li>python310Packages.pgmpy.dist</li>
    <li>python310Packages.piano-transcription-inference</li>
    <li>python310Packages.piano-transcription-inference.dist</li>
    <li>python310Packages.pydmd</li>
    <li>python310Packages.pydmd.dist</li>
    <li>python310Packages.pynvml</li>
    <li>python310Packages.pynvml.dist</li>
    <li>python310Packages.pyro-ppl</li>
    <li>python310Packages.pyro-ppl.dist</li>
    <li>python310Packages.resize-right</li>
    <li>python310Packages.resize-right.dist</li>
    <li>python310Packages.rising</li>
    <li>python310Packages.rising.dist</li>
    <li>python310Packages.safetensors</li>
    <li>python310Packages.safetensors.dist</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.skorch</li>
    <li>python310Packages.skorch.dist</li>
    <li>python310Packages.slicer</li>
    <li>python310Packages.slicer.dist</li>
    <li>python310Packages.snorkel</li>
    <li>python310Packages.snorkel.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-lookups-data</li>
    <li>python310Packages.spacy-lookups-data.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>python310Packages.speechbrain</li>
    <li>python310Packages.speechbrain.dist</li>
    <li>python310Packages.stanza</li>
    <li>python310Packages.stanza.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.tables</li>
    <li>python310Packages.tables.dist</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.test-tube.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.torch (python310Packages.torchWithCuda ,python310Packages.torchWithoutRocm)</li>
    <li>python310Packages.torch.dev (python310Packages.torchWithCuda.dev ,python310Packages.torchWithoutRocm.dev)</li>
    <li>python310Packages.torch.dist (python310Packages.torchWithCuda.dist ,python310Packages.torchWithoutRocm.dist)</li>
    <li>python310Packages.torch.lib (python310Packages.torchWithCuda.lib ,python310Packages.torchWithoutRocm.lib)</li>
    <li>python310Packages.torchaudio</li>
    <li>python310Packages.torchaudio.dist</li>
    <li>python310Packages.torchdiffeq</li>
    <li>python310Packages.torchdiffeq.dist</li>
    <li>python310Packages.torchgpipe</li>
    <li>python310Packages.torchgpipe.dist</li>
    <li>python310Packages.torchio</li>
    <li>python310Packages.torchio.dist</li>
    <li>python310Packages.torchlibrosa</li>
    <li>python310Packages.torchlibrosa.dist</li>
    <li>python310Packages.torchmetrics</li>
    <li>python310Packages.torchmetrics.dist</li>
    <li>python310Packages.torchsde</li>
    <li>python310Packages.torchsde.dist</li>
    <li>python310Packages.transformers</li>
    <li>python310Packages.transformers.dist</li>
    <li>python310Packages.translatehtml</li>
    <li>python310Packages.translatehtml.dist</li>
    <li>python310Packages.ttach</li>
    <li>python310Packages.ttach.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.bitsandbytes</li>
    <li>python311Packages.bitsandbytes.dist</li>
    <li>python311Packages.blosc2</li>
    <li>python311Packages.blosc2.dist</li>
    <li>python311Packages.deepdish</li>
    <li>python311Packages.deepdish.dist</li>
    <li>python311Packages.deepwave</li>
    <li>python311Packages.deepwave.dist</li>
    <li>python311Packages.experiment-utilities</li>
    <li>python311Packages.experiment-utilities.dist</li>
    <li>python311Packages.ezyrb</li>
    <li>python311Packages.ezyrb.dist</li>
    <li>python311Packages.fairscale</li>
    <li>python311Packages.fairscale.dist</li>
    <li>python311Packages.flammkuchen</li>
    <li>python311Packages.flammkuchen.dist</li>
    <li>python311Packages.flyingsquid</li>
    <li>python311Packages.flyingsquid.dist</li>
    <li>python311Packages.fvcore</li>
    <li>python311Packages.fvcore.dist</li>
    <li>python311Packages.gpytorch</li>
    <li>python311Packages.gpytorch.dist</li>
    <li>python311Packages.ignite</li>
    <li>python311Packages.ignite.dist</li>
    <li>python311Packages.iopath</li>
    <li>python311Packages.iopath.dist</li>
    <li>python311Packages.kornia</li>
    <li>python311Packages.kornia.dist</li>
    <li>python311Packages.layoutparser</li>
    <li>python311Packages.layoutparser.dist</li>
    <li>python311Packages.linear_operator</li>
    <li>python311Packages.linear_operator.dist</li>
    <li>python311Packages.lion-pytorch</li>
    <li>python311Packages.lion-pytorch.dist</li>
    <li>python311Packages.monai</li>
    <li>python311Packages.monai.dist</li>
    <li>python311Packages.pandas-stubs</li>
    <li>python311Packages.pandas-stubs.dist</li>
    <li>python311Packages.pdfplumber</li>
    <li>python311Packages.pdfplumber.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.pgmpy</li>
    <li>python311Packages.pgmpy.dist</li>
    <li>python311Packages.pydmd</li>
    <li>python311Packages.pydmd.dist</li>
    <li>python311Packages.pynvml</li>
    <li>python311Packages.pynvml.dist</li>
    <li>python311Packages.pyro-ppl</li>
    <li>python311Packages.pyro-ppl.dist</li>
    <li>python311Packages.resize-right</li>
    <li>python311Packages.resize-right.dist</li>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python311Packages.safetensors</li>
    <li>python311Packages.safetensors.dist</li>
    <li>python311Packages.slicer</li>
    <li>python311Packages.slicer.dist</li>
    <li>python311Packages.snorkel</li>
    <li>python311Packages.snorkel.dist</li>
    <li>python311Packages.speechbrain</li>
    <li>python311Packages.speechbrain.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.tables</li>
    <li>python311Packages.tables.dist</li>
    <li>python311Packages.test-tube</li>
    <li>python311Packages.test-tube.dist</li>
    <li>python311Packages.torch (python311Packages.torchWithCuda ,python311Packages.torchWithoutRocm)</li>
    <li>python311Packages.torch.dev (python311Packages.torchWithCuda.dev ,python311Packages.torchWithoutRocm.dev)</li>
    <li>python311Packages.torch.dist (python311Packages.torchWithCuda.dist ,python311Packages.torchWithoutRocm.dist)</li>
    <li>python311Packages.torch.lib (python311Packages.torchWithCuda.lib ,python311Packages.torchWithoutRocm.lib)</li>
    <li>python311Packages.torchaudio</li>
    <li>python311Packages.torchaudio.dist</li>
    <li>python311Packages.torchdiffeq</li>
    <li>python311Packages.torchdiffeq.dist</li>
    <li>python311Packages.torchgpipe</li>
    <li>python311Packages.torchgpipe.dist</li>
    <li>python311Packages.torchio</li>
    <li>python311Packages.torchio.dist</li>
    <li>python311Packages.torchmetrics</li>
    <li>python311Packages.torchmetrics.dist</li>
    <li>python311Packages.torchsde</li>
    <li>python311Packages.torchsde.dist</li>
    <li>python311Packages.transformers</li>
    <li>python311Packages.transformers.dist</li>
    <li>python311Packages.ttach</li>
    <li>python311Packages.ttach.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
    <li>xpraWithNvenc</li>
    <li>xpraWithNvenc.dist</li>
  </ul>
</details>

**Comment by SomeoneSerge** [comment]
###### Failed derivations

<details>


- [ ] python3.10-torchvision-0.15.2.drv (derivation hash: i6zfiswg0z27kfbfg4rqav2lnf4qv1vh)
  - https://gist.github.com/SomeoneSerge/7d43dd7d4fd988a955d411be29c64800
- [ ] python3.10-invisible-watermark-0.1.5.drv (derivation hash: mvx2l46w7c0zyvp2v1zg6cp139d2kjrn)
  - https://gist.github.com/SomeoneSerge/5c90df4269fb16dbaf8d41b576ad7c1a
- [ ] python3.10-skrl-0.10.2.drv (derivation hash: zppaipi31ilv6i78bv7s7f8blli1ffxi)
  - https://gist.github.com/SomeoneSerge/c24b70ca8a4fb9e1953a3ff3b2e07a2f
- [ ] tiny-cuda-nn-1.6.drv (derivation hash: gv90vni69cns9zfny9yp53j6s01phc03)
  - https://gist.github.com/SomeoneSerge/95b1ac21b6b10d72365e1a394202ff4b
- [ ] python3.11-pyemd-0.5.1.drv (derivation hash: w94v71j3af5bp36iwyc9bz5xk0k06l21)
  - https://gist.github.com/SomeoneSerge/d35de4a0f112057a8b1a05b22cf6e1a6
- [ ] python3.10-xformers-0.0.20.drv (derivation hash: jrm2r2273smri0d0nplp0i3phcq5b62x)
  - https://gist.github.com/SomeoneSerge/a01e2da80d4c43f7121020d9c793c7cd
- [ ] python3.11-invisible-watermark-0.1.5.drv (derivation hash: 1klnplwaay7p8014w21x7294snzvd8nr)
  - https://gist.github.com/SomeoneSerge/566a3ac9b98d73741b7583bcfdfc1d48
- [ ] python3.11-nianet-1.1.4.drv (derivation hash: hxb63iwiqv2brrblxw5m9w9phfx17xkr)
  - https://gist.github.com/SomeoneSerge/931ee6e61e5def335654aa895fec1728
- [ ] python3.11-skorch-0.14.0.drv (derivation hash: 677nh8p278fgszg68ib0vkv3gxszr4k5)
  - https://gist.github.com/SomeoneSerge/533c9828fa35f70d799051329e5e1e7b
- [ ] python3.11-torchvision-0.15.2.drv (derivation hash: ss6bkjjnzh89x78zil2my00npk622nvb)
  - https://gist.github.com/SomeoneSerge/c79a2307f763eaf9c243c673b24f475c
- [ ] python3.11-mmengine-0.7.4.drv (derivation hash: fqrnldy6ws52h3ijsck5ylmi163yp9l6)
  - https://gist.github.com/SomeoneSerge/4dc393121716783a767d1c80a747cef2
- [ ] python3.11-xformers-0.0.20.drv (derivation hash: 2dvd2yvqf4f4p6885jw65s5p5pr4m3ml)
  - https://gist.github.com/SomeoneSerge/68f34fd5cbb8f569a73ee9edaff8c547
- [ ] python3.11-fairseq-0.12.3.drv (derivation hash: 2sliqjxlmm6vgw1y7riffinnhfzyc95c)
  - https://gist.github.com/SomeoneSerge/333ff96d58f88cdc579c72e7c1f92750
- [ ] tiny-cuda-nn-1.6.drv (derivation hash: rigwvxr222rjzdd04fxpmc5gqz2ivddj)
  - https://gist.github.com/SomeoneSerge/650c2be8794c01b47828b16645f2e04a
- [ ] python3.11-apptools-5.1.0.drv (derivation hash: p4gqd3bkw9zfa90sbf3kcixhgx3ffa57)
  - https://gist.github.com/SomeoneSerge/90652284d3d06b9fb18cafc4eb72e1a1
- [ ] tensorflow-gpu-2.11.1.drv (derivation hash: mlp58wmshxqsr2f3253kxdldhqk52n8w)
  - https://gist.github.com/SomeoneSerge/62ddba7beb8882a771339d9a1463f0dd
- [ ] python3.11-botorch-0.8.5.drv (derivation hash: a1l283glh0xz7frysiwd1f7dpzcwqr41)
  - https://gist.github.com/SomeoneSerge/cbabb2cc379c012cd9abc3d284111c52


</details>

**Comment by bcdarwin** [comment]
I think a lot of these are new failures on master due to `python311Packages.tensorboard` having been enabled (maybe should have targeted `staging`).   `python310Packages.torchvision` is probably the most concerning one (failing with CUDA-related errors).

**Comment by SomeoneSerge** [comment]
RE: torchvision


Unbelievavble. This actually resulted in a WITH_CUDA=OFF build. The setup py even seems to ignore USE_CUDA=1. We really should invoke cmake directly

**Comment by SomeoneSerge** [comment]
```
cmake_dependent_option(
    USE_CUDNN "Use cuDNN" ON
    "USE_CUDA" OFF)
```


This silently resets use_cuda to false. There's a bunch of other occasions where pytorch does that. The behaviour we'd like whenwe've set with_cuda but a dependency is missing is immediate failure

**Comment by SomeoneSerge** [comment]
Looking at the logs, this torch revision seems to ignore cmake_cudatoolkit_root. Maybe the release is still using the legacy findcuda. I guess I'll just come back to this in 2 weeks

**Comment by SomeoneSerge** [comment]
Superseded by the much more thorough implementation in https://github.com/NixOS/nixpkgs/pull/249259!

**Pull request closed without merging**